lib/puppet/pops/loader/base_loader.rb

```diff
@@ -145,11 +145,12 @@ def format_uri(uri)
   # 3. find it here
   # 4. give up
   #
-  def internal_load(typed_name)
+  def internal_load(typed_name, action = nil)
```

Comment:
It does seem like this logic fundamentally depends on the plan not already being loaded to behave correctly.

I think I'm generally concerned by the concept of adding "action" to the loader. It seems more natural for "plan_object" to be a loadable thing or for us to make ":plan" more than a closure over the actual plan.